components/atoms/details2/index.js

```diff
@@ -0,0 +1,188 @@
+
+  useEffect(() => {
+    if (unmountContentOnClose) {
+      IsOpen
+        ? setChildrenMounted(() => true)
+        : setTimeout(() => {
+            window &&
+              window.requestAnimationFrame(() => {
+                setChildrenMounted(() => false)
+              })
+          }, animationDuration ?? 300)
+    }
+    setTimeout(
+      () => {
+        window &&
+          window.requestAnimationFrame(() => {
+            contentRef.current && setContentHeight(() => contentRef.current.offsetHeight)
+          })
+      },
+      !IsOpen ? animationDuration ?? 300 : 0
+    )
+
+    return () => {}
+  }, [contentRef.current, IsOpen])
+
+  const handleEventRecieve = e => {
+    setIsOpen(e)
+  }
+
+  useComponentCommunication({
+    id: uniqueId,
+    messageName: MessageNames.DetailsComponent.SET_OPEN,
+    onRecieve: e => handleEventRecieve(e),
+    subscriber: DetailsSubscriber
+  })
+
+  const handleOpenClose = () => {
+    setIsOpen(IsOpen => !IsOpen)
+  }
+
+  return (
+    <Wrapper context={context} {...props}>
+      <Header>
+        <OpenIconAndTitleWrapper>
+          {(content || children) && (
+            <OpenCloseWrapper
+              open={IsOpen}
+              onClick={handleOpenClose}
+              animationDuration={animationDuration}
+            >
+              {icon ? <Icon icon={icon} prefix='fas' /> : <CaretRight />}{' '}
+            </OpenCloseWrapper>
+          )}
+          <Text size={SIZE.MD} style={{ fontWeight: '600' }} context='blackText' content={title} />
+        </OpenIconAndTitleWrapper>
+        <ActionsWrapper>
+          <StartActionComponentWrapper>
+            {startActionComponent} {(toolbar || endActionComponent) && <ActionsDivider />}
+          </StartActionComponentWrapper>
+          <ToolbarWrapper>
+            {toolbar}
+            {endActionComponent && <ActionsDivider />}
+          </ToolbarWrapper>
+          {endActionComponent}
+        </ActionsWrapper>
+      </Header>
+      {(content || children) && (
+        <ContentWrapper
+          maxHeight={ContentHeight}
+          open={IsOpen}
+          animationDuration={animationDuration}
+        >
+          <Content ref={contentRef}>{ChildrenMounted && (content || children)}</Content>
+        </ContentWrapper>
+      )}
+    </Wrapper>
+  )
+}
+
+const CaretRight = styled.div`
+  border-bottom: 5px solid transparent;
+  border-left: 5px solid ${({ theme }) => theme.COLOUR.blackText};
+  border-top: 5px solid transparent;
+  height: 0;
+  width: 0;
+`
+
+const ActionsDivider = styled.div`
+  background-color: ${({ theme }) => theme.COLOUR.darkGrey};
+  height: 1rem;
+  margin: 0 1rem;
+  width: 2px;
+`
+const ToolbarWrapper = styled.div`
+  align-items: center;
+  display: flex;
+`
+const StartActionComponentWrapper = styled.div`
+  align-items: center;
+  display: flex;
+`
+
+const ActionsWrapper = styled.div`
+  align-items: center;
+  display: flex;
+`
+const Content = styled.div`
+  padding: 1rem 0;
```

Comment:
Why not any horizontal paddings for content?

Reply:
i let that for the user to decide, vertical padding is to make sure all of the content is visible

Reply:
@mohsenParvar But I think most of users want to have horizontal padding by default. We can control it with a prop.